testsuite/cluster-test/src/cluster_swarm/cluster_swarm_kube.rs

```diff
@@ -188,6 +246,111 @@ impl ClusterSwarmKube {
 
 #[async_trait]
 impl ClusterSwarm for ClusterSwarmKube {
+    async fn add_network_delay(
+        &self,
+        source_instance: Instance,
+        configuration: Vec<(Vec<Instance>, Duration)>,
+    ) -> Result<()> {
+        let mut command = "".to_string();
+        command += "tc qdisc delete dev eth0 root; ";
+        // Create a HTB https://linux.die.net/man/8/tc-htb
+        command += "tc qdisc add dev eth0 root handle 1: htb; ";
+        for i in 0..configuration.len() {
+            // Create a class within the HTB https://linux.die.net/man/8/tc
+            command += format!(
+                "tc class add dev eth0 parent 1: classid 1:{} htb rate 1tbit; ",
+                i + 1
+            )
+            .as_str();
+        }
+        for (i, config) in configuration.iter().enumerate() {
+            // Create u32 filters so that all the target instances are classified as class 1:(i+1)
+            // http://man7.org/linux/man-pages/man8/tc-u32.8.html
+            for target_instance in &config.0 {
+                command += format!("tc filter add dev eth0 parent 1: protocol ip prio 1 u32 flowid 1:{} match ip dst {}; ", i+1, target_instance.ip()).as_str();
+            }
+        }
+        for (i, config) in configuration.iter().enumerate() {
+            // Use netem to delay packets to this class
+            command += format!(
+                "tc qdisc add dev eth0 parent 1:{} handle {}0: netem delay {}ms; ",
+                i + 1,
+                i + 1,
+                config.1.as_millis(),
+            )
+            .as_str();
+        }
+
+        let back_off_limit = 0;
```

Comment:
For all this code below, can we just have some convenience function that would run arbitrary command using arbitrary docker image on host?
Something like `cluster_swarm.run(instance: Instance, docker_image: String, command: String)`.
Caller code can then use join to multiplex those for multiple instances, as this is fairly common pattern.

Reply:
Done